homeassistant/components/broadlink/remote.py

```diff
@@ -434,3 +438,42 @@ async def _async_learn_rf_command(self, command):
             self.hass.components.persistent_notification.async_dismiss(
                 notification_id="learn_command"
             )
+
+    async def async_delete_command(self, **kwargs):
+        """Delete a list of commands from a remote."""
+        kwargs = SERVICE_DELETE_SCHEMA(kwargs)
+        commands = kwargs[ATTR_COMMAND]
+        device = kwargs[ATTR_DEVICE]
+
+        if not self._state:
+            _LOGGER.warning(
+                "remote.delete_command canceled: %s entity is turned off",
+                self.entity_id,
+            )
+            return
+
+        try:
+            codes = self._codes[device]
+        except KeyError:
+            _LOGGER.error("Failed to delete command. Device not found: %s", device)
+            return
+
+        should_store = False
+
+        for command in commands:
+            try:
+                del codes[command]
+            except KeyError:
+                _LOGGER.error(
```

Comment:
We should `raise` here so the user gets feedback that they had invalid input.

Reply:
This service can receive a list of commands to be deleted. Perhaps some are valid and some are not. I think it is better not to raise the exception in this case, so we can exclude all valid ones.

Reply:
Or we can raise only if no commands were found. Yes, I think this is the way to go. I'll do it.